helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ClusterStatusMonitor.java

```diff
@@ -55,6 +55,81 @@
 import org.slf4j.LoggerFactory;
 
 public class ClusterStatusMonitor implements ClusterStatusMonitorMBean {
+  private class AsyncMissingTopStateMonitor extends Thread {
```

Comment:
This will be meaningful if our pipeline speed is very fast. Otherwise, it will not help as the cache updated state is refreshed per pipeline. I dont believe we need build this thread but just rely on pipeline call as we did before. 

But we need to remove the constraint of doing only final reporting.

We can discuss about it tomorrow f2f.